validate.js

```diff
@@ -0,0 +1,57 @@
+const bb = require('ssb-bendy-butt')
+const bfe = require('ssb-bfe')
+
+exports.validateSingle = function (contentSection) {
+  if (!Array.isArray(contentSection) || contentSection.length !== 2)
+    return new Error(
+      `invalid message: contentSection ${typeof contentSection} with length ${
+        contentSection.length
+      } is incorrect, expected a list of content and contentSignature`
+    )
+
+  const [content, contentSignature] = contentSection
+
+  if (!Array.isArray(content) || content.length < 3 || content.length > 4)
```

Comment:
you can have really any kind of metadata on the feed. So instead of checking these lengths I think we should just check that if this is an add then it at least has the required fields, same with tombstone.